src/octoprint/server/api/access.py

```diff
@@ -163,6 +163,8 @@ def add_user():
         userManager.add_user(name, password, active, permissions, groups)
     except users.UserAlreadyExists:
         abort(409)
+    except users.InvalidUserName:
+        abort(409)
```

Comment:
HTTP 409 is not the correct status code here, because that means that there is a conflict. With username already existing, there is a conflict with the existing user, so 409.

For invalid username, I would say maybe 400 (bad request), but I'm happy if there is something better suited for that to be suggested.

```suggestion
        abort(400, "Username invalid")
```

If something other than 400 is used, then the API documentation would have to be updated as well.

Reply:
Resolved